app/controllers/user_controller.rb

```diff
@@ -53,7 +56,7 @@ def user_params
   end
 
   def check_authorized_access
-    raise CanCan::AccessDenied.new("Unauthorized access!", :read, User) unless ((@user.admin_id == current_user.id) && (@user.id != current_user.id) && (@user.role != "Admin"))
+    raise CanCan::AccessDenied.new("Unauthorized access!", :read, User) unless ((@user.admin_id == current_user.id) && (@user.id != current_user.id) && (@user.role != "admin"))
```

Comment:
As I understand only Admin user can create other user. For that you just need to check `current_user.admin?`.